src/cascadia/UnitTests_TerminalCore/ConptyRoundtripTests.cpp

```diff
@@ -0,0 +1,367 @@
+
+    VERIFY_SUCCEEDED(renderer.PaintFrame());
+}
+
+// Function Description:
+// - Helper function to validate that a number of characters in a row are all
+//   the same. Validates that the next end-start characters are all equal to the
+//   provided string. Will move the provided iterator as it validates. The
+//   caller should ensure that `iter` starts where they would like to validate.
+// Arguments:
+// - expectedChar: The character (or characters) we're expecting
+// - iter: a iterator pointing to the cell we'd like to start validating at.
+// - start: the first index in the range we'd like to validate
+// - end: the last index in the range we'd like to validate
+// Return Value:
+// - <none>
+void _verifySpanOfText(const wchar_t* const expectedChar,
+                       TextBufferCellIterator& iter,
+                       const int start,
+                       const int end)
+{
+    for (int x = start; x < end; x++)
+    {
+        SetVerifyOutput settings(VerifyOutputSettings::LogOnlyFailures);
+        if (iter->Chars() != expectedChar)
+        {
+            Log::Comment(NoThrowString().Format(L"character [%d] was mismatched", x));
+        }
+        VERIFY_ARE_EQUAL(expectedChar, (iter++)->Chars());
+    }
+    Log::Comment(NoThrowString().Format(
+        L"Successfully validated %d characters were '%s'", end - start, expectedChar));
+}
+
+void ConptyRoundtripTests::ConptyOutputTestCanary()
+{
+    Log::Comment(NoThrowString().Format(
+        L"This is a simple test to make sure that everything is working as expected."));
+    VERIFY_IS_NOT_NULL(_pVtRenderEngine.get());
+
+    _flushFirstFrame();
+}
+
+void ConptyRoundtripTests::SimpleWriteOutputTest()
+{
+    Log::Comment(NoThrowString().Format(
+        L"Write some simple output, and make sure it gets rendered largely "
+        L"unmodified to the terminal"));
+    VERIFY_IS_NOT_NULL(_pVtRenderEngine.get());
+
+    auto& g = ServiceLocator::LocateGlobals();
+    auto& renderer = *g.pRender;
+    auto& gci = g.getConsoleInformation();
+    auto& si = gci.GetActiveOutputBuffer();
+    auto& hostSm = si.GetStateMachine();
+    auto& termTb = *term->_buffer;
+
+    _flushFirstFrame();
+
+    expectedOutput.push_back("Hello World");
+    hostSm.ProcessString(L"Hello World");
+
+    VERIFY_SUCCEEDED(renderer.PaintFrame());
+
+    auto iter = termTb.GetCellDataAt({ 0, 0 });
+    VERIFY_ARE_EQUAL(L"H", (iter++)->Chars());
+    VERIFY_ARE_EQUAL(L"e", (iter++)->Chars());
+    VERIFY_ARE_EQUAL(L"l", (iter++)->Chars());
+    VERIFY_ARE_EQUAL(L"l", (iter++)->Chars());
+    VERIFY_ARE_EQUAL(L"o", (iter++)->Chars());
+    VERIFY_ARE_EQUAL(L" ", (iter++)->Chars());
+    VERIFY_ARE_EQUAL(L"W", (iter++)->Chars());
+    VERIFY_ARE_EQUAL(L"o", (iter++)->Chars());
+    VERIFY_ARE_EQUAL(L"r", (iter++)->Chars());
+    VERIFY_ARE_EQUAL(L"l", (iter++)->Chars());
+    VERIFY_ARE_EQUAL(L"d", (iter++)->Chars());
+    VERIFY_ARE_EQUAL(L" ", (iter++)->Chars());

```

Comment:
Can you please put this into some sort of helper that operates on the expected string and the iterator since you use this pattern over and over?

Reply:
I guess it's not exactly the same use below... but it still feels like it should be some sort of helper.